category/vm/compiler/ir/x86/emitter.cpp

```diff
@@ -1228,7 +1228,7 @@ namespace monad::vm::compiler::native
         return is_live(reg, live, std::index_sequence_for<LiveSet...>{});
     }
 
-    void Emitter::gas_decrement_no_check(int32_t gas)
+    void Emitter::gas_decrement_no_check(int64_t gas)
```

Comment:
If `gas > std::numeric_limits<int32_t>::max()` we need to do something else. For example,
```
as_.mov(x86::rax, gas);
gas_decrement_no_check(x86::rax);
```
This only works if `x86::rax` is not live everywhere this is called (directly or indirectly), and I have not checked this.

Reply:
Remember to add a unit test for this, or open an issue.

Reply:
Well, `x86::rax` can be live, just not live at the same time as `gas > INT32_MAX`